components/popover_list_members/index.ts

```diff
@@ -2,27 +2,50 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
 
 import {createSelector} from 'reselect';
 
 import {getAllChannelStats} from 'mattermost-redux/selectors/entities/channels';
 import {getCurrentRelativeTeamUrl} from 'mattermost-redux/selectors/entities/teams';
 import {getCurrentUserId, getUserStatuses, makeGetProfilesInChannel} from 'mattermost-redux/selectors/entities/users';
 import {getTeammateNameDisplaySetting, getAddMembersToChannel} from 'mattermost-redux/selectors/entities/preferences';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Channel} from 'mattermost-redux/types/channels';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+
+import {GlobalState} from 'types/store';
+import {ModalData} from 'types/actions';
 
 import {openDirectChannelToUserId} from 'actions/channel_actions.jsx';
 import {loadProfilesAndStatusesInChannel} from 'actions/user_actions.jsx';
 import {openModal} from 'actions/views/modals';
+
 import {canManageMembers} from 'utils/channel_utils';
 import {sortUsersByStatusAndDisplayName} from 'utils/utils.jsx';
 
-import PopoverListMembers from './popover_list_members.jsx';
+import PopoverListMembers, {Props} from './popover_list_members';
+
+type Filters = {
+    role?: string;
+    inactive?: boolean;
+    active?: boolean;
+    roles?: string[];
+    exclude_roles?: string[];
+    channel_roles?: string[];
+    team_roles?: string[];
+};
+
+type Actions = {
+    openModal: <P>(modalData: ModalData<P>) => void;
+    loadProfilesAndStatusesInChannel: (channelId: string, page: number, perPage: number | undefined, sort: string, options: {active: boolean}) => void;
+    openDirectChannelToUserId: (userId: string) => Promise<{data: Channel}>;
+}
 
-const makeSortUsersByStatusAndDisplayName = (doGetProfilesInChannel) => {
+const makeSortUsersByStatusAndDisplayName = (doGetProfilesInChannel: (state: GlobalState, channelId: Channel['id'], filters?: Filters) => UserProfile[]) => {
     return createSelector(
         'makeSortUsersByStatusAndDisplayName',
-        (state, channelId) => doGetProfilesInChannel(state, channelId, true),
+        (state: GlobalState, channelId: string) => doGetProfilesInChannel(state, channelId, {active: true}),
```

Comment:
not really needed, but to be on the safe side we could add `Channel['id']` here as type as well instead of string:

```suggestion
        (state: GlobalState, channelId: Channel['id']) => doGetProfilesInChannel(state, channelId, {active: true}),
```